File: 136. Single Number/single number.py
__author__ = 'Bill Quan'
"""
    Given an array of integers, every element appears twice except for one. Find that single one.
"""

class Solution(object):
    def singleNumber(self, nums):
        """

        :param nums: list[int]
        :return: int
        """
        if len(nums) == 0:
            return 0
        res = nums[0]
        i = 1 
        while i < len(nums):
            res ^= nums[i]
            i += 1
        return res

File: 136. Single Number/single number.cpp
class Solution
{
public:
	int singleNumber(int[] nums, int n)
	{
		if (n == 0)
			return 0;
		int res = 0;
		for (int i = 0; i < n; i++)
		{
			res ^= nums[i];  // xor by bits 
		}
		return res;
	}
};